TeensyDmx.cpp

```diff
@@ -1021,14 +1436,52 @@ void TeensyDmx::respondMessage(uint16_t nackReason)
         m_rdmBuffer.dataLength = 2;
         putUInt16(&m_rdmBuffer.data, nackReason);
     }
-    m_rdmBuffer.length = m_rdmBuffer.dataLength + RDM_PACKET_SIZE_NO_PD;  // total packet length
 
     ++m_rdmBuffer.cmdClass;
 
+    sendRDMMessage();
+}
+
+
+void TeensyDmx::buildSendRDMMessage(byte *uid, uint8_t commandClass, uint16_t pid) {
+    if (m_rdm != nullptr) {
+        m_rdmBuffer.startCode = E120_SC_RDM;
+        m_rdmBuffer.subStartCode = E120_SC_SUB_MESSAGE;
+
+        memcpy(m_rdmBuffer.destId, uid, RDM_UID_LENGTH);
+        memcpy(m_rdmBuffer.sourceId, m_rdm->uid, RDM_UID_LENGTH);
+
+        // Sub Dev
+        putUInt16(&m_rdmBuffer.subDev, RDM_ROOT_DEVICE);
+
+        m_rdmBuffer.messageCount = 0; // Number of queued messages
+        m_rdmBuffer.responseType = 1; // Port 1
+        m_rdmBuffer.transNo = 0;
+        // Parameter
+        putUInt16(&m_rdmBuffer.parameter, pid);
+
+        m_rdmBuffer.cmdClass = commandClass;
+
+        sendRDMMessage();
+        m_controllerState = ControllerState::RDM_MESSAGE;
+        // TODO(Peter): Set to RDM_BROADCAST if dest is broadcast
+    }
+}
+
+
+void TeensyDmx::sendRDMMessage()
+{
+    m_state = IDLE;
+    // no need to set these data fields:
+    // StartCode, SubStartCode
+
+    m_rdmBuffer.length = m_rdmBuffer.dataLength + RDM_PACKET_SIZE_NO_PD;  // total packet length
+
     uint16_t checkSum = rdmCalculateChecksum(reinterpret_cast<uint8_t*>(&m_rdmBuffer),
                                              m_rdmBuffer.length);
 
     // Send reply
+    stopTransmit();
     stopReceive();
     setDirection(true);
```

Comment:
I do wonder if an enum : bool would be better for setDirection, this isn't exactly clear.

Reply:
Perhaps setDirectionTransmit(bool) is easier and clearer?

Reply:
Possibly, I might do that as a separate thing later, just wanted the discussion really :)